blazar_dashboard/content/leases/templates/leases/_widget_capabilities.html

```diff
@@ -146,15 +179,32 @@
         values.forEach(function(cv) {
             element.appendChild(new Option(cv, cv));
         });
-        setDefaultNodeType();
+        setResourcePropertyValues(element.id.slice(-1));
     }
 
     function setDefaultNodeType(){
         var nodeOptions = $("#resource-node-type");
-        console.log("getting node options in widget capabilitiesjs")
         try {
           nodeOptions.find('[value="compute_haswell"]').attr("selected","selected");
-          console.log("node options in widget capabilitiesjs found: " + nodeOptions);
+        } catch(err) {
+          console.error(err);
+        }
+    }
+
+    function setPropertyType(index, propType){
+        var nodeOptions = $('#criteria-resource_properties-value-' + index);
+        try {
+          nodeOptions.find('[value="' + propType + '"]').attr("selected","selected");
```

Comment:
How can this throw an exception? I thought jQuery would just return a selector of length 0, and then things like `find` and `attr` would effectively be no-ops.

Reply:
I think that's also left over from a refactor, i'll clean it up